### DISCUSSION TITLE
Looking for new features or refinements

### DISCUSSION THREAD

**Discussion by josephrcox** [josephrcox/easy-reddit-downloader#78] [Q&A]
**Hey all!** 👋

I am planning a code refactor of this project, and would love to hear if there are new features or refinements that would 10x your experience with the tool. 

Let me know here! 

Best,
Joseph

**Comment by bhugh65** [comment]
would love to see settings in the user_config that can restrict or imposes pre defined limits to on demand queries made.
like limit downloads to "x", or automatically set other options to an alternate user default so they do not have to be typed unless needed. unless it can be already done and I am missing something. 

thanks to you for this great project,
Brian

**Comment by Jalae** [comment]
i don't see it in the readme and havn't looked at the code but a way to download the discussions on other media post types - ie someone posts a picture of an issue as an image post with title "help" all i get is the picture, but the discussion in the comments is actually the content i'd want.

**Reply by Jalae** [reply]
made my own modification to downloader.js to grab the comments for other media types in a side file using the same output code for the self/text files. This probably isn't how you would do it (no config flag to enable disable for example) and there may be bugs but it's how I got it working on my end. 
[downloaders.js](https://github.com/user-attachments/files/26152746/downloaders.js)

**Reply by josephrcox** [reply]
@Jalae Can you give a Reddit post example please? Then I can look into adding this officially. 
- Post URL
- What exactly you are looking to have downloaded

**Reply by Jalae** [reply]
ok so for https://old.reddit.com/r/plan9/comments/1efumce/surprisingly_it_boots_mouse_works_but_thats_about/
only the picture is grabbed. 
but i'd also want the discussion on the page. With my modification I also get the attached file in the same directory as the picture
[2024-07-30_plan9_plan9deck_surprisingly it boots, mouse works but thats about it.txt](https://github.com/user-attachments/files/26160522/2024-07-30_plan9_plan9deck_surprisingly.it.boots.mouse.works.but.thats.about.it.txt)

**Reply by Jalae** [reply]
similarly for https://old.reddit.com/r/plan9/comments/mw208b/plan_9_desktop_guide/ this only gets an html that links to the page, none of the comments.  (this is basically adding the process of creating a self/text file for all other media types, but using the permalink_url for cases where the url goes off reddit)
